Tests/FoundationEssentialsTests/GregorianCalendarTests.swift

```diff
@@ -18,6 +18,44 @@ import TestSupport
 @testable import FoundationEssentials
 #endif
 
+extension DateComponents {
+    init(field: Calendar.Component, value: Int) {
```

Comment:
Hah - I added this in my other PR here:

https://github.com/apple/swift-foundation/pull/322/files#diff-10a4fcffe1d9245a1a5371a1ce086a0a67726b8dfde2454781613fecea34469bR82

```swift
    package init?(component: Calendar.Component, value: Int) {
        switch component {
        case .calendar, .timeZone, .isLeapMonth:
            return nil
        default:
            setValue(value, for: component)
        }
    }
```